src/auspex/exp_factory.py

```diff
@@ -153,18 +153,26 @@ class QubitExpFactory(object):
 
     @staticmethod
     def run(meta_file=None, meas_file=None, expname=None, calibration=False, save_data=True,
-           cw_mode=False, repeats=None, single_plotter=True):
+           cw_mode=False, repeats=None, single_plotter=auspex.config.single_plotter_mode):
         """This passes all of the parameters given to the *create* method
         and then runs the experiment immediately."""
+
+        # allow the plotter mode to be overridden
+        spm = auspex.config.single_plotter_mode
+        auspex.config.single_plotter_mode = single_plotter
+
         exp = QubitExpFactory.create(meta_file=meta_file, meas_file=meas_file, expname=expname,
                                      calibration=calibration, cw_mode=cw_mode, save_data=save_data,
-                                    repeats=repeats, single_plotter=single_plotter)
+                                    repeats=repeats)
         exp.run_sweeps()
+
+        # set the plotting mode back to the default value
+        auspex.config.single_plotter_mode = spm
         return exp
 
     @staticmethod
     def create(meta_file=None, meas_file=None, expname=None, calibration=False, save_data=True,
-               cw_mode=False, instr_filter=None, repeats=None, single_plotter=True):
```

Comment:
This way the convenience method doesn't work if you create the experiment and subsequently add loops.
```python
exp = exp_factory.create()
exp.add_sweep(...)
exp.run_sweeps()
```
Is that okay?

Reply:
That or have `run_sweeps` coded like `run` with a save settings, update, run, load saved settings?